src/panels/lovelace/components/hui-marquee.ts

```diff
@@ -0,0 +1,102 @@
+import {
+  html,
+  LitElement,
+  PropertyValues,
+  TemplateResult,
+  customElement,
+  css,
+  CSSResult,
+  property,
+} from "lit-element";
+
+const marqueeSpeed = 0.2;
+
+@customElement("hui-marquee")
+class HuiMarquee extends LitElement {
+  @property() public marqueeText?: string;
+  @property() public off?: boolean;
+  private interval?: number;
+  private left: number = 0;
+
+  protected updated(changedProperties: PropertyValues): void {
+    super.updated(changedProperties);
+
+    if (!this.off) {
+      this._start();
+    }
+  }
+
+  protected render(): TemplateResult {
+    if (!this.marqueeText) {
+      return html``;
+    }
+
+    return html`
+      <div id="Marquee">
+        <div>${this.marqueeText}</div>
+      </div>
+    `;
+  }
+
+  private get _marqueeElement(): HTMLElement {
+    return this.shadowRoot?.querySelector("#Marquee") as HTMLElement;
+  }
+
+  private get _marqueeElementFirstChild(): HTMLElement {
+    return this._marqueeElement.firstElementChild as HTMLElement;
+  }
+
+  private get _marqueeElementLastChild(): HTMLElement {
+    return this._marqueeElement.lastElementChild as HTMLElement;
+  }
+
+  private _start(): void {
+    if (this.interval || this.offsetWidth >= this._marqueeElement.scrollWidth) {
+      return;
+    }
+
+    this._marqueeElement.innerHTML += this._marqueeElement.innerHTML;
+
+    this.interval = window.setInterval(() => {
+      this._play();
+    });
+  }
+
+  private _play(): void {
+    this._marqueeElement.style.marginLeft = "-" + this.left + "px";
+
+    if (this.off && !this.left) {
```

Comment:
Shouldn't this logic be moved to `updated` ?

Reply:
No. Because it is in an interval this can run without being updated. We want the text to continue to scroll until back at its original position. Therefore we check each time it loops